diff --git a/bin/npx b/bin/npx
index 30c3df13b7d40712e122d4d8291666bba1b0982e..6b7cfe7059739695ee97172467f39437684cafa4 100755
--- a/bin/npx
+++ b/bin/npx
@@ -4,8 +4,13 @@
 # Yarn and npm.
 
 # shellcheck disable=SC2154
-if [[ ! "$npm_config_user_agent" =~ yarn ]] && [ -x "$(command -v yarn)" ]; then
+# Note that this script is already merged in expo
+# Waiting for the next release of expo-modules-script to be published
+# https://github.com/expo/expo/pull/37976/files
+if [[ "$npm_config_user_agent" =~ yarn ]] && [ -x "$(command -v yarn)" ]; then
   yarn exec -- npx "$@"
+elif [[ "$npm_config_user_agent" =~ pnpm ]] && [ -x "$(command -v pnpm)" ]; then
+  pnpm exec -- npx "$@"
 else
   npx "$@"
 fi
