diff --git a/babel.js b/babel.js
index 5a8a29ba204e6a2bd4a74d4a30be2329bf94eaf5..d79f57460356f97e69f6470ab0d91d19e5bad8d5 100644
--- a/babel.js
+++ b/babel.js
@@ -9,7 +9,7 @@ module.exports = function () {
           importSource: "react-native-css-interop",
         },
       ],
-      "react-native-reanimated/plugin",
+      "react-native-worklets/plugin",
     ],
   };
 };
diff --git a/dist/runtime/components.js b/dist/runtime/components.js
index 8835a2d9705857c11ed70b242266f9790542d408..3aeff79f92c3aaf3fb77e79e64a0241cf800dca2 100644
--- a/dist/runtime/components.js
+++ b/dist/runtime/components.js
@@ -5,7 +5,7 @@ const react_native_1 = require("react-native");
 const api_1 = require("./api");
 (0, api_1.cssInterop)(react_native_1.Image, { className: "style" });
 (0, api_1.cssInterop)(react_native_1.Pressable, { className: "style" });
-(0, api_1.cssInterop)(react_native_1.SafeAreaView, { className: "style" });
+// (0, api_1.cssInterop)(react_native_1.SafeAreaView, { className: "style" });
 (0, api_1.cssInterop)(react_native_1.Switch, { className: "style" });
 (0, api_1.cssInterop)(react_native_1.Text, { className: "style" });
 (0, api_1.cssInterop)(react_native_1.TouchableHighlight, { className: "style" });
@@ -46,11 +46,11 @@ const api_1 = require("./api");
     ListHeaderComponentClassName: "ListHeaderComponentStyle",
     contentContainerClassName: "contentContainerStyle",
 });
-try {
-    const SafeAreaView = require("react-native-safe-area-context").SafeAreaView;
-    (0, api_1.cssInterop)(SafeAreaView, {
-        className: "style",
-    });
-}
-catch { }
+// try {
+//     const SafeAreaView = require("react-native-safe-area-context").SafeAreaView;
+//     (0, api_1.cssInterop)(SafeAreaView, {
+//         className: "style",
+//     });
+// }
+// catch { }
 //# sourceMappingURL=components.js.map
\ No newline at end of file
diff --git a/dist/runtime/native/api.js b/dist/runtime/native/api.js
index 5ea02d81fd0954009978e04b12542e58a954af7e..58e05868a02600a10f06f175fe02b4f7555dcb05 100644
--- a/dist/runtime/native/api.js
+++ b/dist/runtime/native/api.js
@@ -24,7 +24,7 @@ const cssInterop = (baseComponent, mapping) => {
     const type = (0, unwrap_components_1.getComponentType)(baseComponent);
     if (type === "function") {
         component = (props) => {
-            return (0, native_interop_1.interop)(baseComponent, configs, props, undefined);
+            return (0, native_interop_1.interop)(baseComponent, configs, props, props.ref)
         };
     }
     else {
